docs/software/communication/nccl.md

```diff
@@ -4,7 +4,78 @@
 [NCCL](https://developer.nvidia.com/nccl) is an optimized inter-GPU communication library for NVIDIA GPUs.
 It is commonly used in machine learning frameworks, but traditional scientific applications can also benefit from NCCL.
 
-!!! todo
-    - high level description
-    - libfabric/aws-ofi-nccl plugin
-    - configuration options
+## Using NCCL
+
+To use the Slingshot network on Alps, the [`aws-ofi-nccl`](https://github.com/aws/aws-ofi-nccl) plugin must be used.
+With the container engine, the [AWS OFI NCCL hook][ref-ce-aws-ofi-hook] can be used to load the plugin into the container and configure NCCL to use it.
+
+Most uenvs, like [`prgenv-gnu`][ref-uenv-prgenv-gnu], also contain the NCCL plugin.
+When using e.g. the `default` view of `prgenv-gnu` the `aws-ofi-nccl` plugin will be available in the environment.
+Alternatively, loading the `aws-ofi-nccl` module with the `modules` view also makes the plugin available in the environment.
+The environment variables described below must still be set to ensure that NCCL uses the plugin.
+
+While the container engine sets these automatically when using the NCCL hook, the following environment variables should always be set for correctness and optimal performance when using NCCL:
+
+```bash
+export NCCL_NET="AWS Libfabric" # (1)
+export NCCL_NET_GDR_LEVEL=PHB # (2)
+export FI_CXI_DEFAULT_CQ_SIZE=131072 # (3)
+export FI_CXI_DEFAULT_TX_SIZE=32768
+export FI_CXI_DISABLE_HOST_REGISTER=1
+export FI_CXI_RX_MATCH_MODE=software
+export FI_MR_CACHE_MONITOR=userfaultfd
+export MPICH_GPU_SUPPORT_ENABLED=0 # (4)
```

Comment:
```suggestion
export NCCL_NET="AWS Libfabric" # (1)!
export NCCL_NET_GDR_LEVEL=PHB # (2)!
export FI_CXI_DEFAULT_CQ_SIZE=131072 # (3)!
export FI_CXI_DEFAULT_TX_SIZE=32768
export FI_CXI_DISABLE_HOST_REGISTER=1
export FI_CXI_RX_MATCH_MODE=software
export FI_MR_CACHE_MONITOR=userfaultfd
export MPICH_GPU_SUPPORT_ENABLED=0 # (4)!
```
Hide the `#` symbol.

Reply:
Thanks for remembering! I'll add a note about this to the style guide as well.